Automattic/sensei
# Add file watch & live browser reload options to Gulp

#### Changes proposed in this Pull Request:

* Add file watch and browser live reload capabilities to Gulp to speed up development.
* Site needs to be accessed through a BrowserSync server this spins up for the live reload to work
* Also added dotenv for easy configuration as developers might be running local Wordpress on different ports/urls

#### Testing instructions:

* Update dev dependencies with `npm install`
* Create an .env file with WORDPRESS_HOST= url you use to access Wordpress (e.g localhost + Docker container's public port)
* Run `gulp serve`
* Open browser at http://localhost:8242
* Make changes to .scss files => Updated CSS appears in browser
* Make changes to JS files => Browser reloads after recompiling js 

#### Issues:

* Wordpress seem to pick up that it's now accessed on a different port, both frontend and wp-admin works through the BrowserSync proxy
* But Customizer does not, makes calls to the proxied URL, causing CSP issues
